diff --git a/lib/Basic/Targets.cpp b/lib/Basic/Targets.cpp
index 064c54b..f5ad131 100644
--- a/lib/Basic/Targets.cpp
+++ b/lib/Basic/Targets.cpp
@@ -1995,6 +1995,7 @@ void X86TargetInfo::getDefaultFeatures(llvm::StringMap<bool> &Features) const {
     setFeatureEnabled(Features, "sse4.1", true);
     break;
   case CK_Atom:
+    setFeatureEnabled(Features, "mmx", true);
     setFeatureEnabled(Features, "ssse3", true);
     break;
   case CK_Corei7:
