From: Hazou

Fixes build of libcorkscrew for x86 devices

diff --git a/libcorkscrew/arch-x86/backtrace-x86.c b/libcorkscrew/arch-x86/backtrace-x86.c
index 9c4428c..1677999 100755
--- a/libcorkscrew/arch-x86/backtrace-x86.c
+++ b/libcorkscrew/arch-x86/backtrace-x86.c
@@ -18,13 +18,6 @@
  * Backtracing functions for x86.
  */
 
-#ifndef __BIONIC__
-// glibc has its own renaming of the Linux kernel's structures.
-// define this here, so that it's defined before the first inclusion
-// of glibc's features.h
-#  define _GNU_SOURCE // For REG_EBP, REG_ESP, and REG_EIP.
-#endif
-
 #define LOG_TAG "Corkscrew"
 //#define LOG_NDEBUG 0
 
@@ -90,6 +83,7 @@ typedef struct ucontext {
 #else /* ! __BIONIC__ && ! __APPLE__ */
 
 /* _GNU_SOURCE is already defined above */
+#define __USE_GNU // For REG_EBP, REG_ESP, and REG_EIP.
 #  include <ucontext.h>
 
 #endif
