diff --git a/healthd/healthd_board_default.cpp b/healthd/healthd_board_default.cpp
index 024b44f..b2bb516 100644
--- a/healthd/healthd_board_default.cpp
+++ b/healthd/healthd_board_default.cpp
@@ -25,5 +25,5 @@ void healthd_board_init(struct healthd_config *config)
 int healthd_board_battery_update(struct android::BatteryProperties *props)
 {
     // return 0 to log periodic polled battery status to kernel log
-    return 1;
+    return 0;
 }
diff --git a/init/init.c b/init/init.c
index a33a265..ecfcfd3 100644
--- a/init/init.c
+++ b/init/init.c
@@ -1303,7 +1303,7 @@ int main(int argc, char **argv)
             continue;
 
         for (i = 0; i < fd_count; i++) {
-            if (ufds[i].revents & POLLIN) {
+            if (ufds[i].revents == POLLIN) {
                 if (ufds[i].fd == get_property_set_fd())
                     handle_property_set_fd();
                 else if (ufds[i].fd == get_keychord_fd())
diff --git a/init/init.h b/init/init.h
index db94c29..408779f 100644
--- a/init/init.h
+++ b/init/init.h
@@ -76,7 +76,7 @@ struct svcenvinfo {
 #define SVC_RESTART     0x100 /* Use to safely restart (stop, wait, start) a service */
 
 #ifndef NR_SVC_SUPP_GIDS
-#define NR_SVC_SUPP_GIDS 12    /* twelve supplementary groups */
+#define NR_SVC_SUPP_GIDS 28   /* twentyeight supplementary groups */
 #endif
 
 #define COMMAND_RETRY_TIMEOUT 5
diff --git a/init/ueventd.c b/init/ueventd.c
index b973185..53cc9ac 100644
--- a/init/ueventd.c
+++ b/init/ueventd.c
@@ -99,7 +99,7 @@ int ueventd_main(int argc, char **argv)
         nr = poll(&ufd, 1, -1);
         if (nr <= 0)
             continue;
-        if (ufd.revents & POLLIN)
+        if (ufd.revents == POLLIN)
                handle_device_fd();
     }
 }
diff --git a/init/util.c b/init/util.c
index f9216ea..635bb65 100644
--- a/init/util.c
+++ b/init/util.c
@@ -409,6 +409,10 @@ void get_hardware_name(char *hardware, unsigned int *revision)
     int fd, n;
     char *x, *hw, *rev;
 
+    /* Hardware string was provided on kernel command line */
+    if (hardware[0])
+        return;
+
     fd = open("/proc/cpuinfo", O_RDONLY);
     if (fd < 0) return;
 
diff --git a/libsuspend/autosuspend.c b/libsuspend/autosuspend.c
index eb1f66e..7083dd0 100644
--- a/libsuspend/autosuspend.c
+++ b/libsuspend/autosuspend.c
@@ -48,10 +48,9 @@ static int autosuspend_init(void)
         goto out;
     }
 
-    if (!autosuspend_ops) {
-        ALOGE("failed to initialize autosuspend\n");
-        return -1;
-    }
+    autosuspend_inited = false;
+    ALOGE("failed to initialize autosuspend\n");
+    return -1;
 
 out:
     autosuspend_inited = true;
diff --git a/libsuspend/autosuspend_earlysuspend.c b/libsuspend/autosuspend_earlysuspend.c
index 1df8c6a..56310b4 100644
--- a/libsuspend/autosuspend_earlysuspend.c
+++ b/libsuspend/autosuspend_earlysuspend.c
@@ -143,7 +143,6 @@ static int autosuspend_earlysuspend_disable(void)
     if (ret < 0) {
         strerror_r(errno, buf, sizeof(buf));
         ALOGE("Error writing to %s: %s\n", EARLYSUSPEND_SYS_POWER_STATE, buf);
-        goto err;
     }
 
     if (wait_for_earlysuspend) {
@@ -157,9 +156,6 @@ static int autosuspend_earlysuspend_disable(void)
     ALOGV("autosuspend_earlysuspend_disable done\n");
 
     return 0;
-
-err:
-    return ret;
 }
 
 struct autosuspend_ops autosuspend_earlysuspend_ops = {
@@ -208,20 +204,9 @@ struct autosuspend_ops *autosuspend_earlysuspend_init(void)
         return NULL;
     }
 
-    ret = write(sPowerStatefd, "on", 2);
-    if (ret < 0) {
-        strerror_r(errno, buf, sizeof(buf));
-        ALOGW("Error writing 'on' to %s: %s\n", EARLYSUSPEND_SYS_POWER_STATE, buf);
-        goto err_write;
-    }
-
     ALOGI("Selected early suspend\n");
 
     start_earlysuspend_thread();
 
     return &autosuspend_earlysuspend_ops;
-
-err_write:
-    close(sPowerStatefd);
-    return NULL;
 }
diff --git a/toolbox/Android.mk b/toolbox/Android.mk
index 3e28494..b4e39ca 100644
--- a/toolbox/Android.mk
+++ b/toolbox/Android.mk
@@ -72,9 +72,9 @@ TOOLS := \
 	mkswap \
 	readlink
 
-ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
+#ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
 TOOLS += r
-endif
+#endif
 
 TOOLS += setfattr
 
