diff --git a/minui/Android.mk b/minui/Android.mk
index ae587d8..77fed0a 100644
--- a/minui/Android.mk
+++ b/minui/Android.mk
@@ -45,6 +45,7 @@ LOCAL_SRC_FILES := $(common_src_files)
 LOCAL_ADDITIONAL_DEPENDENCIES := $(common_additional_dependencies)
 LOCAL_C_INCLUDES += $(common_c_includes)
 LOCAL_SHARED_LIBRARIES := libpng libpixelflinger
+LOCAL_SHARED_LIBRARIES_x86 := libenc
 LOCAL_CFLAGS += $(common_cflags) -DSHARED_MINUI
 LOCAL_WHOLE_STATIC_LIBRARIES := $(common_whole_static_libraries)
 include $(BUILD_SHARED_LIBRARY)
