diff --git a/videocodec/Android.mk b/videocodec/Android.mk
index 6a79dd0..1655e2e 100644
--- a/videocodec/Android.mk
+++ b/videocodec/Android.mk
@@ -42,7 +42,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderAVC.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderAVC
 ifeq ($(TARGET_BOARD_PLATFORM),clovertrail)
@@ -104,7 +104,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderVP8.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderVP8
 
@@ -176,7 +176,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderVP9HWR.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderVP9HWR
 
@@ -223,7 +223,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderVP9Hybrid.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderVP9Hybrid
 
@@ -278,7 +278,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderMPEG4.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderMPEG4
 ifeq ($(TARGET_BOARD_PLATFORM),clovertrail)
@@ -335,7 +335,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderH263.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderH263
 ifeq ($(TARGET_BOARD_PLATFORM),clovertrail)
@@ -392,7 +392,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderWMV.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderWMV
 ifeq ($(TARGET_BOARD_PLATFORM),clovertrail)
@@ -457,7 +457,7 @@ LOCAL_CFLAGS += -DVED_TILING
 LOCAL_SHARED_LIBRARIES += libdl
 endif
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderAVCSecure
 
@@ -498,7 +498,7 @@ LOCAL_SRC_FILES := \
     OMXVideoEncoderAVC.cpp
 
 LOCAL_CFLAGS += $(LOCAL_C_FLAGS)
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoEncoderAVC
 include $(BUILD_SHARED_LIBRARY)
@@ -540,7 +540,7 @@ LOCAL_CFLAGS += $(LOCAL_C_FLAGS)
 ifeq ($(SW_MPEG4_ENCODER),true)
     LOCAL_CFLAGS += -DSYNC_MODE
 endif
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoEncoderH263
 include $(BUILD_SHARED_LIBRARY)
@@ -579,7 +579,7 @@ LOCAL_SRC_FILES := \
 
 LOCAL_CFLAGS += $(LOCAL_C_FLAGS)
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoEncoderMPEG4
 include $(BUILD_SHARED_LIBRARY)
@@ -612,7 +612,7 @@ LOCAL_SRC_FILES := \
     OMXVideoDecoderBase.cpp \
     OMXVideoDecoderPAVC.cpp
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoDecoderPAVC
 ifeq ($(TARGET_BOARD_PLATFORM),clovertrail)
@@ -662,7 +662,7 @@ LOCAL_SRC_FILES := \
     OMXVideoEncoderVP8.cpp
 
 LOCAL_CFLAGS += $(LOCAL_C_FLAGS)
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := libOMXVideoEncoderVP8
 include $(BUILD_SHARED_LIBRARY)
diff --git a/videocodec/OMXVideoDecoderBase.h b/videocodec/OMXVideoDecoderBase.h
index 48706b8..49610a7 100644
--- a/videocodec/OMXVideoDecoderBase.h
+++ b/videocodec/OMXVideoDecoderBase.h
@@ -27,7 +27,7 @@
 #include "graphics.h"
 #endif
 
-static constexpr const char* VA_VED_RAW_MIME_TYPE = "video/x-raw-vaved";
+static const char* VA_VED_RAW_MIME_TYPE = "video/x-raw-vaved";
 static const uint32_t VA_VED_COLOR_FORMAT = 0x20;
 
 
diff --git a/videocodec/libvpx_internal/libvpx.mk b/videocodec/libvpx_internal/libvpx.mk
index 5a95803..7df2470 100644
--- a/videocodec/libvpx_internal/libvpx.mk
+++ b/videocodec/libvpx_internal/libvpx.mk
@@ -50,7 +50,7 @@ LOCAL_CFLAGS += -Wno-unused-function
 # in vp[89]_[cd]x_iface.c
 LOCAL_CLANG_CFLAGS += -Wno-missing-field-initializers
 
-LOCAL_CFLAGS += -Werror
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE := libvpx_internal
 
 LOCAL_MODULE_CLASS := STATIC_LIBRARIES
@@ -123,7 +123,7 @@ LOCAL_GENERATED_SOURCES += $(VPX_GEN)
 
 ifeq ($(libvpx_target),x86)
 libvpx_x86_asm_src_files := $(filter %.asm, $(libvpx_codec_srcs_unique))
-libvpx_x86_yasm_dir := prebuilts/misc/$(HOST_OS)-$(HOST_PREBUILT_ARCH)/yasm
+libvpx_x86_yasm_dir := /usr/bin
 
 X86_ASM_GEN := $(addprefix $(libvpx_intermediates)/, $(libvpx_x86_asm_src_files:%.asm=%.asm.o))
 
