diff --git a/minui/Android.mk b/minui/Android.mk
index 5e42d2f..2967e63 100644
--- a/minui/Android.mk
+++ b/minui/Android.mk
@@ -30,6 +30,7 @@ else
 endif
 
 LOCAL_STATIC_LIBRARY := libpng
+LOCAL_STATIC_LIBRARY_x86 := libenc
 LOCAL_WHOLE_STATIC_LIBRARIES := libpixelflinger_static
 LOCAL_MODULE := libminui
 
@@ -97,6 +98,7 @@ LOCAL_MODULE := libminui
 
 LOCAL_ARM_MODE:= arm
 LOCAL_SHARED_LIBRARIES := libpng libpixelflinger
+LOCAL_SHARED_LIBRARIES_x86 := libenc
 # This used to compare against values in double-quotes (which are just
 # ordinary characters in this context).  Strip double-quotes from the
 # value so that either will work.
diff --git a/minuitwrp/Android.mk b/minuitwrp/Android.mk
index c9749eb..605bc97 100644
--- a/minuitwrp/Android.mk
+++ b/minuitwrp/Android.mk
@@ -143,6 +143,7 @@ endif
 LOCAL_CFLAGS += -DTWRES=\"$(TWRES_PATH)\"
 LOCAL_SHARED_LIBRARIES += libz libc libcutils libjpeg libpng
 LOCAL_STATIC_LIBRARIES += libpixelflinger_static
+LOCAL_STATIC_LIBRARIES_x86 := libenc
 LOCAL_MODULE_TAGS := eng
 LOCAL_MODULE := libminuitwrp
 
