diff --git a/src/Android.mk b/src/Android.mk
index 3ec4397..676c640 100644
--- a/src/Android.mk
+++ b/src/Android.mk
@@ -39,8 +39,7 @@ LOCAL_CLANG_CFLAGS += \
 
 LOCAL_CFLAGS := \
     -DLINUX -DANDROID -g -Wall -Wno-unused \
-    -DPSBVIDEO_LOG_ENABLE -DPSBVIDEO_VXD392 \
-    -DPSBVIDEO_MSVDX_DEC_TILING -DPSBVIDEO_MSVDX_EC
+    -DPSBVIDEO_LOG_ENABLE
 
 LOCAL_C_INCLUDES := \
     $(call include-path-for, libhardware)/hardware \
@@ -84,6 +83,9 @@ LOCAL_SRC_FILES := \
     tng_jpegdec.c
 
 ifneq ($(filter $(TARGET_BOARD_PLATFORM),merrifield moorefield morganfield),)
+LOCAL_CFLAGS += -DPSBVIDEO_VXD392 \
+    -DPSBVIDEO_MSVDX_DEC_TILING -DPSBVIDEO_MSVDX_EC
+
 LOCAL_SRC_FILES += \
     pnw_H263ES.c \
     pnw_H264ES.c \
@@ -141,7 +143,24 @@ LOCAL_CFLAGS += -DPSBVIDEO_MRFL_VPP_ROTATE
 endif
 endif
 
+else ifeq ($(TARGET_BOARD_PLATFORM),sc1) # Medfield mfld_pr2 (Razr I)
+LOCAL_SRC_FILES += \
+    psb_texture.c \
+    pnw_H263ES.c \
+    pnw_H264ES.c \
+    pnw_MPEG4ES.c \
+    pnw_cmdbuf.c \
+    pnw_hostcode.c \
+    pnw_hostheader.c \
+    pnw_hostjpeg.c \
+    pnw_jpeg.c \
+    tng_ved_scaling.c \
+    psb_ws_driver.c
+LOCAL_CFLAGS += -DPSBVIDEO_MFLD
+LOCAL_SHARED_LIBRARIES += libpvr2d
+
 else
+
 LOCAL_CFLAGS += \
     -DPSBVIDEO_VXD392 -DBAYTRAIL \
     -DPSBVIDEO_MSVDX_DEC_TILING -DPSBVIDEO_MSVDX_EC
@@ -152,7 +171,8 @@ LOCAL_SRC_FILES += android/psb_mds.cpp
 LOCAL_CFLAGS += -DTARGET_HAS_MULTIPLE_DISPLAY
 LOCAL_SHARED_LIBRARIES += libmultidisplay
 endif
-LOCAL_CFLAGS += -Werror
+
+#LOCAL_CFLAGS += -Werror
 LOCAL_MODULE_TAGS := optional
 LOCAL_MODULE := pvr_drv_video
 
