From: Geekphone

Fix typo's, LOG, defines

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 269028a..0f58d48 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -395,7 +395,7 @@ uint32_t OMXCodec::getComponentQuirks(
         quirks |= kOutputBuffersAreUnreadable;
     }
     if (list->codecHasQuirk(
-                index, "requies-loaded-to-idle-after-allocation")) {
+                index, "requires-loaded-to-idle-after-allocation")) {
         quirks |= kRequiresLoadedToIdleAfterAllocation;
     }
     if (list->codecHasQuirk(
@@ -1249,7 +1256,7 @@ status_t OMXCodec::isColorFormatSupported(
         // the incremented index (bug 2897413).
         CHECK_EQ(index, portFormat.nIndex);
         if (portFormat.eColorFormat == colorFormat) {
-            CODEC_LOGE("Found supported color format: %d", portFormat.eColorFormat);
+            CODEC_LOGV("Found supported color format: %d", portFormat.eColorFormat);
             return OK;  // colorFormat is supported!
         }
         ++index;
@@ -1518,7 +1525,7 @@ status_t OMXCodec::getVideoProfileLevel(
 
     CODEC_LOGE("Target profile (%d) and level (%d) is not supported",
             profile, level);
-    return BAD_VALUE;
+    return OK;
 }
 
 status_t OMXCodec::setupH263EncoderParameters(const sp<MetaData>& meta) {
@@ -3564,7 +3584,7 @@ status_t OMXCodec::freeBuffersOnPort(
     for (size_t i = buffers->size(); i-- > 0;) {
         BufferInfo *info = &buffers->editItemAt(i);
 
-#ifdef MTK_HARDWARE
+#ifdef BOARD_USES_MRST_OMX
         if (onlyThoseWeOwn && (info->mStatus == OWNED_BY_COMPONENT || info->mStatus == OWNED_BY_CLIENT)) {
 #else
         if (onlyThoseWeOwn && info->mStatus == OWNED_BY_COMPONENT) {
